gateway-js/src/index.ts

```diff
@@ -727,13 +793,40 @@ export class ApolloGateway implements GraphQLService {
       });
     }
 
+    const canUseManagedConfig =
+      this.apolloConfig?.graphId && this.apolloConfig?.keyHash;
     if (!canUseManagedConfig) {
       throw new Error(
         'When `serviceList` is not set, an Apollo configuration must be provided. See https://www.apollographql.com/docs/apollo-server/federation/managed-federation/ for more information.',
```

Comment:
Error isn't quite right, since there are other approaches like csdl

Reply:
Also added a test for this: 238fbec

Open to suggestions on rewording this, this one's kinda weird and, as demonstrated here, not the most obvious to keep up to date.